### PR TITLE
[Merged by Bors] - Enable logging

### PR DESCRIPTION
**Pull request by maltesander**
# Description

- Added log4j2 logging for Hive

If logging is set to DEBUG, there will be some exceptions due to null values in the XML string from `DataNucleus`.

```
ERROR StatusLogger com.fasterxml.jackson.core.JsonGenerationException: Invalid null character in text to output
 com.fasterxml.jackson.core.JsonGenerationException: Invalid null character in text to output
     at com.fasterxml.jackson.dataformat.xml.util.StaxUtil.throwAsGenerationException(StaxUtil.java:51)
```

Printed null values causing the problem (without XML format):
```
bash-4.4$ sed -n '/\x0/ { s/\x0/<NUL>/g; p}' /stackable/log/test.log
[DEBUG] 2023-01-20 13:33:24.637 [main] Transaction - Transaction created [DataNucleus Transaction, ID=Xid=<NUL><NUL><NUL>, enlisted resources=[]]
[DEBUG] 2023-01-20 13:33:24.647 [main] Transaction - Running enlist operation on resource: org.datanucleus.store.rdbms.ConnectionFactoryImpl$EmulatedXAResource@687e4c93, error code TMNOFLAGS and transaction: [DataNucleus Transaction, ID=Xid=<NUL><NUL><NUL>, enlisted resources=[]]
[DEBUG] 2023-01-20 13:33:24.647 [main] Connection - ManagedConnection(enlisted) "org.datanucleus.store.rdbms.ConnectionFactoryImpl$ManagedConnectionImpl@3b1ed14b [conn=com.zaxxer.hikari.pool.HikariProxyConnection@3727f0ee, commitOnRelease=false, closeOnRelease=false, closeOnTxnEnd=true]" starting for transaction "Xid=<NUL><NUL><NUL>" with flags "0"
[DEBUG] 2023-01-20 13:33:24.698 [main] Transaction - Committing [DataNucleus Transaction, ID=Xid=<NUL><NUL><NUL>, enlisted resources=[org.datanucleus.store.rdbms.ConnectionFactoryImpl$EmulatedXAResource@687e4c93]]
[DEBUG] 2023-01-20 13:33:24.698 [main] Connection - ManagedConnection(enlisted) 
```
## TODO

- [x] Adapt image version in examples, docs, tests 
- [x] Adapt changelog

fixes: https://github.com/stackabletech/hive-operator/issues/290

## Review Checklist

- [x] Code contains useful comments
- [x] CRD change approved (or not applicable)
- [x] (Integration-)Test cases added (or not applicable)
- [x] Documentation added (or not applicable)
- [x] Changelog updated (or not applicable)
- [x] Cargo.toml only contains references to git tags (not specific commits or branches)
- [x] Helm chart can be installed and deployed operator works (or not applicable)

Once the review is done, comment `bors r+` (or `bors merge`) to merge. [Further information](https://bors.tech/documentation/getting-started/#reviewing-pull-requests)

### CODE REVIEW

**Comment by maltesander** [comment]
test: https://ci.stackable.tech/view/02%20Operator%20Tests%20(custom)/job/hive-operator-it-custom/23/

**Review by siegfriedweber — CHANGES_REQUESTED** [review]

File: rust/operator-binary/src/product_logging.rs [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+    #[snafu(display("crd validation failure"))]
+    CrdValidationFailure { source: stackable_hive_crd::Error },
+    #[snafu(display("vectorAggregatorConfigMapName must be set"))]
+    MissingVectorAggregatorAddress,
+}
+
+type Result<T, E = Error> = std::result::Result<T, E>;
+
+const VECTOR_AGGREGATOR_CM_ENTRY: &str = "ADDRESS";
+const CONSOLE_CONVERSION_PATTERN: &str = "%d{ISO8601} %5p [%t] %c{2}: %m%n";
```
[the remark this answers is not in the document]
Comment (maltesander) [reply rust/operator-binary/src/product_logging.rs:40]: I used the pattern that is shipped with hive 2.3.9 and 3.1.3
https://github.com/apache/hive/blob/branch-2.3/data/conf/hive-log4j2.properties
https://github.com/apache/hive/blob/branch-3.1/data/conf/hive-log4j2.properties

File: tests/templates/kuttl/logging/00-install-vector-aggregator-discovery-configmap.yaml.j2 [hunk 2]
```diff
@@ -0,0 +1,9 @@
+{% if lookup('env', 'VECTOR_AGGREGATOR') %}
```
[the remark this answers is not in the document]
Comment (maltesander) [reply tests/templates/kuttl/logging/00-install-vector-aggregator-discovery-configmap.yaml.j2:1]: https://github.com/stackabletech/hive-operator/pull/298/commits/ef114f550f2b77fcc29fc72ba4c0419a40122365

File: rust/crd/src/lib.rs [hunk 3]
```diff
@@ -14,26 +14,33 @@ use stackable_operator::{
...
+// directories
 pub const STACKABLE_CONFIG_DIR: &str = "/stackable/config";
-pub const STACKABLE_RW_CONFIG_DIR: &str = "/stackable/rwconfig";
+pub const STACKABLE_CONFIG_DIR_NAME: &str = "config";
+pub const STACKABLE_CONFIG_MOUNT_DIR: &str = "/stackable/mount/config";
+pub const STACKABLE_CONFIG_MOUNT_DIR_NAME: &str = "config-mount";
+pub const STACKABLE_LOG_DIR: &str = "/stackable/log";
+pub const STACKABLE_LOG_DIR_NAME: &str = "log";
+pub const STACKABLE_LOG_MOUNT_DIR: &str = "/stackable/mount/log";
+pub const STACKABLE_LOG_MOUNT_DIR_NAME: &str = "log-mount";
```
[the remark this answers is not in the document]
Comment (maltesander) [reply rust/crd/src/lib.rs]: done https://github.com/stackabletech/hive-operator/pull/298/commits/0d7925e74cda1e2bff2777df304c50e986ab77ae

File: tests/templates/kuttl/resources/00-install-vector-aggregator-discovery-configmap.yaml.j2 [hunk 4]
```diff
@@ -0,0 +1,9 @@
+{% if lookup('env', 'VECTOR_AGGREGATOR') %}
+---
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: vector-aggregator-discovery
```
Comment (siegfriedweber) [inline tests/templates/kuttl/resources/00-install-vector-aggregator-discovery-configmap.yaml.j2:6]: The vector-aggregator-discovery ConfigMap must also be referenced in the cluster specs of all integration tests and the Vector agent must be enabled, e.g.:

```yaml
---
apiVersion: hive.stackable.tech/v1alpha1
kind: HiveCluster
metadata:
  name: test-hive
spec:
  image:
    productVersion: "{{ test_scenario['values']['hive'].split('-stackable')[0] }}"
    stackableVersion: "{{ test_scenario['values']['hive'].split('-stackable')[1] }}"
  clusterConfig:
{% if lookup('env', 'VECTOR_AGGREGATOR') %}
    vectorAggregatorConfigMapName: vector-aggregator-discovery
{% endif %}
  metastore:
    config:
      logging:
        enableVectorAgent: {{ lookup('env', 'VECTOR_AGGREGATOR') | length > 0 }}
    roleGroups:
      default:
        replicas: 1
```

**Review by siegfriedweber — CHANGES_REQUESTED** [review]

File: tests/templates/kuttl/resources/00-install-vector-aggregator-discovery-configmap.yaml.j2 (hunk 4 above)
Comment (maltesander) [reply tests/templates/kuttl/resources/00-install-vector-aggregator-discovery-configmap.yaml.j2:6]: done https://github.com/stackabletech/hive-operator/pull/298/commits/f01b629d36a7587d7f21b8e352b9ce693733af34

**Review by siegfriedweber — APPROVED** [review]
LGTM

**Comment by maltesander** [comment]
bors merge

**Pull request closed without merging**